src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitter.java

```diff
@@ -0,0 +1,193 @@
+
+        AttendanceCourse course = attendanceCourseService.findByCanvasCourseId(courseId);
+
+        Map<Long, String> studentCommentsMap = new HashMap<>();
+        if (course.getShowNotesToStudents()) {
+            studentCommentsMap = attendanceService.getAttendanceCommentsBySectionId(model.getSectionId());
+        }
+
+        //Generates the map with the information to be submitted to Canvas
+        for (AttendanceSummaryModel.Entry entry : model.getEntries()) {
+            studentMap.put(Long.toString(entry.getCanvasStudentId()),
+                    generateStudentSubmissionOptions(isSimpleAttendance, submissionOptions, attendanceAssignment, course, studentCommentsMap, entry));
+        }
+
+        //Pushing the information to Canvas
+        Optional<Progress> returnedProgress;
+        submissionOptions.setStudentSubmissionOptionMap(studentMap);
+        try {
+            returnedProgress = canvasApiWrapperService.gradeMultipleSubmissionsBySection(oauthToken, submissionOptions);
+        } catch (IOException e) {
+            LOG.error("Error while pushing the grades of section: " + model.getSectionId(), e);
+            throw new IOException("Could not push grades of section: " + model.getSectionId());
+        }
+
+        if (!isValidProgress(returnedProgress)) {
+            LOG.error("Error object returned while pushing the grades of section: " + model.getSectionId());
+            throw new IOException("Could not push grades of section: " + model.getSectionId());
+        }
+    }
+
+    /**
+     * Generates the parameters and data needed for push grades or comments to canvas
+     */
+    private MultipleSubmissionsOptions.StudentSubmissionOption generateStudentSubmissionOptions(boolean isSimpleAttendance, MultipleSubmissionsOptions submissionOptions,
+                                                                                               AttendanceAssignment attendanceAssignment, AttendanceCourse course, Map<Long, String> studentCommentsMap, AttendanceSummaryModel.Entry entry) {
+
+        Double grade = calculateStudentGrade(attendanceAssignment, entry, isSimpleAttendance);
+        StringBuilder comments = new StringBuilder(getCommentHeader(entry, course, isSimpleAttendance));
+
+        if (course.getShowNotesToStudents() && studentCommentsMap.keySet().contains(entry.getStudentId())) {
+            comments.append("\nAdditional Comments: \n");
+            comments.append(studentCommentsMap.get(entry.getStudentId()));
+        }
+
+        return submissionOptions.createStudentSubmissionOption(comments.toString(), grade.toString(), null, null, null, null);
+    }
+
+    private boolean isValidProgress(Optional<Progress> returnedProgress) {
+        return (returnedProgress.isPresent() && !returnedProgress.get().getWorkflowState().equals("failed"));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AUselessParenthesesCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "failed" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1132)